pocs/utils/logger.py

```diff
@@ -15,30 +19,153 @@
 all_loggers = {}
 
 
+def field_name_to_key(field_name):
+    """Given a field_name from Formatter.parse(), extract the argument key."""
+    assert isinstance(field_name, str)
+    assert len(field_name)
+    m = re.match(r'^([^.[]+)', field_name)
+    if not m:
+        return None
+    arg_name = m.group(1)
+    if arg_name.isdigit():
+        pos = int(arg_name)
+        return int(arg_name)
+    else:
+        return arg_name
+
+
+def format_references_keys(fmt, args):
```

Comment:
Maybe `format_has_reference_keys` to be more consistent with method below and to clarify usage?